### PR TITLE
Added Route Protection

### PR DESCRIPTION
**Pull request by AkshatSharma5**

### CODE REVIEW

File: package.json
Comment (AryanGKulkarni) [inline package.json:1]: No need of this lib

use this
``` import Cookies from "js-cookie";```

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.0.0",
     "cmdk": "^0.2.0",
+    "cookies-next": "^4.2.1",
```
Comment (AryanGKulkarni) [inline package.json:34]: No need to use this lib

instead use this
```import Cookies from "js-cookie";```

File: src/components/Sidebar.tsx [hunk 2]
```diff
@@ -46,10 +47,41 @@ const Sidebar = () => {
...
+    setCookie('isRecruiter',true);
+    setCookie('isFaculty',false);
+  }
+
+  },[isAdmin,isRecruiter,isStudent]);
+
+
+
+
+
```
Comment (AryanGKulkarni) [inline src/components/Sidebar.tsx:84]: I hope this was used for testing, remove the changes made in this file

File: src/middleware.ts [hunk 3]
```diff
@@ -2,15 +2,26 @@ import { NextResponse } from "next/server";
...
+  const isRecruiter = false;
+
+  if (isAdmin && (request.url.includes('/recruiter') || request.url.includes('/student'))) {
+    return NextResponse.redirect("http://localhost:3000/login/");
+  }
+  else if (isStudent && (request.url.includes('/recruiter') || request.url.includes('/admin'))) {
+    return NextResponse.redirect("http://localhost:3000/login/");
+  }
+  else if (isRecruiter && (request.url.includes('/student') || request.url.includes('/admin'))) {
+    return NextResponse.redirect("http://localhost:3000/login/");
```
Comment (AryanGKulkarni) [inline src/middleware.ts:24]: write this
```return NextResponse.redirect(new URL("/login", request.url));```

File: src/components/Sidebar.tsx (hunk 2 above)
Comment (AryanGKulkarni) [inline src/components/Sidebar.tsx:84]: I hope this was used for testing, revert these changes

File: src/middleware.ts [hunk 4]
```diff
@@ -2,15 +2,26 @@ import { NextResponse } from "next/server";
 import type { NextRequest } from "next/server";
 
 const protectedRoutes = ["/admin/company", "/admin/students", "/admin/job"];
+// const adminRoutes = ['/admin/companies', '/admin/JAF', '/admin/jobs', '/admin/students'];
+// const studentRoutes = ['/student/jobs', '/student/offCampus', '/student/onCampus', '/student/interviewExperiences', '/student/profile', '/student/resumes'];
+// const recruiterRoutes = ['/recruiter/jaf', '/recruiter/prevjaf'];
```
Comment (AryanGKulkarni) [inline src/middleware.ts:7]: why are these lines commented?

File: src/middleware.ts [hunk 5]
```diff
@@ -2,15 +2,26 @@ import { NextResponse } from "next/server";
...
-  ) {
-    return NextResponse.redirect(new URL("/login", request.url));
+  // const isStudent = request.cookies.get("isStudent");
+  const isStudent = true;
+  // const isAdmin = request.cookies.get('isAdmin');
+  const isAdmin = false;
+  // const isRecruiter = request.cookies.get("isRecruiter");
+  const isRecruiter = false;
+
+  if (isAdmin && (request.url.includes('/recruiter') || request.url.includes('/student'))) {
```
Comment (AryanGKulkarni) [inline src/middleware.ts:17]: Instead of this write

```if(isAdmin && !(request.url.includes('/admin')))```

File: src/middleware.ts [hunk 6]
```diff
@@ -2,15 +2,26 @@ import { NextResponse } from "next/server";
...
+    return NextResponse.redirect("http://localhost:3000/login/");
+  }
+  else if (isStudent && (request.url.includes('/recruiter') || request.url.includes('/admin'))) {
+    return NextResponse.redirect("http://localhost:3000/login/");
+  }
+  else if (isRecruiter && (request.url.includes('/student') || request.url.includes('/admin'))) {
```
Comment (AryanGKulkarni) [inline src/middleware.ts:23]: same

File: src/middleware.ts (hunk 4 above)
Comment (AkshatSharma5) [reply src/middleware.ts:7]: they were used for testing.

File: src/middleware.ts (hunk 5 above)
Comment (AkshatSharma5) [reply src/middleware.ts:17]: fixed

File: src/middleware.ts [hunk 7]
```diff
@@ -2,15 +2,26 @@ import { NextResponse } from "next/server";
...
+  const isStudent = true;
+  // const isAdmin = request.cookies.get('isAdmin');
+  const isAdmin = false;
+  // const isRecruiter = request.cookies.get("isRecruiter");
+  const isRecruiter = false;
+
+  if (isAdmin && (request.url.includes('/recruiter') || request.url.includes('/student'))) {
+    return NextResponse.redirect("http://localhost:3000/login/");
+  }
+  else if (isStudent && (request.url.includes('/recruiter') || request.url.includes('/admin'))) {
```
[the remark this answers is not in the document]
Comment (AkshatSharma5) [reply src/middleware.ts:20]: fixed

File: src/middleware.ts (hunk 6 above)
Comment (AkshatSharma5) [reply src/middleware.ts:23]: fixed

**Comment by mittal-ishaan** [comment]
merge this after @Pradeep-Kumar-Rebbavarapu  merges his pr. and use yarn.

**Review by mittal-ishaan — DISMISSED** [review]

**Comment by mittal-ishaan** [comment]
only @AryanGKulkarni should merge this

**Review by AryanGKulkarni — APPROVED** [review]

**Pull request merged**